src/bin/pg_autoctl/monitor.h

```diff
@@ -14,13 +14,28 @@
 #include "pgsql.h"
 #include "state.h"
 
-
 /* interface to the monitor */
 typedef struct Monitor
 {
 	PGSQL pgsql;
 } Monitor;
 
+typedef struct MonitorConfig
+{
+	/* in-memory configuration related variables */
+	ConfigFilePaths pathnames;
+
+	/* pg_autoctl setup */
+	char nodename[_POSIX_HOST_NAME_MAX];
+
+	/* PostgreSQL setup */
+	char role[NAMEDATALEN];
+
+	/* PostgreSQL setup */
+	PostgresSetup pgSetup;
+} MonitorConfig;
+
+
```

Comment:
Why moving the structure definition to `monitor.h` rather than keeping it in `monitor_config.h` and including `monitor_config.h` in `monitor.c` as you did already? That's what we do for the `keeper...` files and I think we should have the same pattern in both the keeper and the monitor code.